src/controllers/tasks/confirm.sic.code.controller.ts

```diff
@@ -6,32 +6,44 @@ import { lookupSicCodeDescription } from "../../utils/api.enumerations";
 import { CompanyProfile } from "@companieshouse/api-sdk-node/dist/services/company-profile/types";
 import { getCompanyProfile } from "../../services/company.profile.service";
 import { SicCode } from "../../types/sic.code";
-import { RADIO_BUTTON_VALUE } from "../../utils/constants";
+import { RADIO_BUTTON_VALUE, SIC_CODE_ERROR } from "../../utils/constants";
 
 export const get = async (req: Request, res: Response, next: NextFunction) => {
   try {
     const companyNumber = getCompanyNumber(req);
     const backLinkUrl = urlUtils.getUrlWithCompanyNumber(TASK_LIST_PATH, companyNumber);
     const companyProfile: CompanyProfile = await getCompanyProfile(companyNumber);
     const sicCodes: SicCode[] = getSicCodeDetails(companyProfile);
-    return res.render(Templates.SIC, {
-      backLinkUrl,
-      sicCodes
-    });
+    return res.render(Templates.SIC, { backLinkUrl, sicCodes });
   } catch (e) {
     return next(e);
   }
 };
 
-export const post = (req: Request, res: Response) => {
-  const sicButtonValue = req.body.sic;
+export const post = async (req: Request, res: Response, next: NextFunction) => {
+  const sicButtonValue = req.body.sicCodeStatus;
   const companyNumber = getCompanyNumber(req);
 
+  if (sicButtonValue === RADIO_BUTTON_VALUE.YES) {
+    return res.redirect(urlUtils.getUrlWithCompanyNumber(TASK_LIST_PATH, companyNumber));
+  }
+
   if (sicButtonValue === RADIO_BUTTON_VALUE.NO) {
     return res.render(Templates.WRONG_SIC, {
       backLinkUrl: urlUtils.getUrlWithCompanyNumber(SIC_PATH, companyNumber)
     });
   }
+
+  try {
+    const companyProfile: CompanyProfile = await getCompanyProfile(companyNumber);
+    return res.render(Templates.SIC, {
```

Comment:
Doesn't this render require a templateName option for matomo to track them properly? Same for line 32